<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>MissionRouter</class>
 <widget class="QMainWindow" name="MissionRouter">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>516</width>
    <height>462</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>Mission Target Router</string>
  </property>
  <widget class="QWidget" name="centralwidget">
   <layout class="QGridLayout" name="gridLayout">
    <item row="0" column="0" colspan="3">
     <widget class="QTableView" name="tableView">
      <property name="sizePolicy">
       <sizepolicy hsizetype="MinimumExpanding" vsizetype="MinimumExpanding">
        <horstretch>1</horstretch>
        <verstretch>1</verstretch>
       </sizepolicy>
      </property>
      <property name="minimumSize">
       <size>
        <width>444</width>
        <height>203</height>
       </size>
      </property>
     </widget>
    </item>
    <item row="9" column="2">
     <widget class="QPushButton" name="optimizeButton">
      <property name="enabled">
       <bool>false</bool>
      </property>
      <property name="text">
       <string>Optimize Route</string>
      </property>
     </widget>
    </item>
    <item row="4" column="0" colspan="3">
     <widget class="QGroupBox" name="groupBox">
      <property name="title">
       <string>Manage custom stops</string>
      </property>
      <layout class="QHBoxLayout" name="horizontalLayout">
       <property name="leftMargin">
        <number>6</number>
       </property>
       <property name="topMargin">
        <number>6</number>
       </property>
       <property name="rightMargin">
        <number>6</number>
       </property>
       <property name="bottomMargin">
        <number>6</number>
       </property>
       <item>
        <widget class="QLabel" name="label">
         <property name="text">
          <string>System Name:</string>
         </property>
        </widget>
       </item>
       <item>
        <widget class="NoBeepLineEdit" name="customSystem">
         <property name="sizePolicy">
          <sizepolicy hsizetype="Expanding" vsizetype="Fixed">
           <horstretch>1</horstretch>
           <verstretch>0</verstretch>
          </sizepolicy>
         </property>
        </widget>
       </item>
       <item>
        <widget class="QPushButton" name="pushButton">
         <property name="text">
          <string>Add</string>
         </property>
        </widget>
       </item>
       <item>
        <widget class="QPushButton" name="pushButton_2">
         <property name="text">
          <string>Clear</string>
         </property>
        </widget>
       </item>
      </layout>
     </widget>
    </item>
    <item row="2" column="0" colspan="3">
     <widget class="QGroupBox" name="groupBox_2">
      <property name="title">
       <string>Mission destination management</string>
      </property>
      <layout class="QHBoxLayout" name="horizontalLayout_2">
       <property name="leftMargin">
        <number>6</number>
       </property>
       <property name="topMargin">
        <number>6</number>
       </property>
       <property name="rightMargin">
        <number>6</number>
       </property>
       <property name="bottomMargin">
        <number>6</number>
       </property>
       <item>
        <widget class="QCheckBox" name="includeMissionSystems">
         <property name="text">
          <string>Load missions from:</string>
         </property>
        </widget>
       </item>
       <item>
        <widget class="QComboBox" name="commanders">
         <property name="sizePolicy">
          <sizepolicy hsizetype="Preferred" vsizetype="Fixed">
           <horstretch>10</horstretch>
           <verstretch>0</verstretch>
          </sizepolicy>
         </property>
        </widget>
       </item>
       <item>
        <widget class="QPushButton" name="refreshButton">
         <property name="text">
          <string>Refresh</string>
         </property>
        </widget>
       </item>
      </layout>
     </widget>
    </item>
   </layout>
  </widget>
  <widget class="QMenuBar" name="menubar">
   <property name="geometry">
    <rect>
     <x>0</x>
     <y>0</y>
     <width>516</width>
     <height>22</height>
    </rect>
   </property>
  </widget>
  <widget class="QStatusBar" name="statusbar"/>
 </widget>
 <customwidgets>
  <customwidget>
   <class>NoBeepLineEdit</class>
   <extends>QLineEdit</extends>
   <header location="global">src/NoBeepLineEdit.h</header>
  </customwidget>
 </customwidgets>
 <resources/>
 <connections>
  <connection>
   <sender>refreshButton</sender>
   <signal>clicked()</signal>
   <receiver>MissionRouter</receiver>
   <slot>refreshMissions()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>461</x>
     <y>310</y>
    </hint>
    <hint type="destinationlabel">
     <x>467</x>
     <y>15</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>commanders</sender>
   <signal>activated(QString)</signal>
   <receiver>MissionRouter</receiver>
   <slot>updateMissionTable()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>185</x>
     <y>308</y>
    </hint>
    <hint type="destinationlabel">
     <x>467</x>
     <y>19</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>optimizeButton</sender>
   <signal>clicked()</signal>
   <receiver>MissionRouter</receiver>
   <slot>optimizeRoute()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>385</x>
     <y>411</y>
    </hint>
    <hint type="destinationlabel">
     <x>467</x>
     <y>14</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>pushButton</sender>
   <signal>clicked()</signal>
   <receiver>MissionRouter</receiver>
   <slot>addStop()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>420</x>
     <y>374</y>
    </hint>
    <hint type="destinationlabel">
     <x>515</x>
     <y>13</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>pushButton_2</sender>
   <signal>clicked()</signal>
   <receiver>MissionRouter</receiver>
   <slot>clearCustom()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>473</x>
     <y>371</y>
    </hint>
    <hint type="destinationlabel">
     <x>515</x>
     <y>13</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>includeMissionSystems</sender>
   <signal>clicked()</signal>
   <receiver>MissionRouter</receiver>
   <slot>updateMissionTable()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>70</x>
     <y>313</y>
    </hint>
    <hint type="destinationlabel">
     <x>1516</x>
     <y>17</y>
    </hint>
   </hints>
  </connection>
 </connections>
 <slots>
  <slot>refreshMissions()</slot>
  <slot>updateMissionTable()</slot>
  <slot>optimizeRoute()</slot>
  <slot>addStop()</slot>
  <slot>clearCustom()</slot>
 </slots>
</ui>
